src/data_type/mod.rs

```diff
@@ -2285,6 +2285,34 @@ impl DataType {
             (Err(_), Err(_)) => Err(Error::other("No common variant")),
         }
     }
+
+    pub fn lower_bound(&self) -> Option<DataType> {
+        match self {
+            DataType::Boolean(b) => b.max().map(|x| DataType::boolean_min(*x)),
+            DataType::Integer(i) => i.max().map(|x| DataType::integer_min(*x)),
+            DataType::Float(f) => f.max().map(|x| DataType::float_min(*x)),
+            DataType::Text(t) => t.max().map(|x| DataType::text_min(x.to_string())),
+            DataType::Date(d) => d.max().map(|x| DataType::date_min(*x)),
+            DataType::Time(t) => t.max().map(|x| DataType::time_min(*x)),
+            DataType::DateTime(d) => d.max().map(|x| DataType::date_time_min(*x)),
+            DataType::Duration(d) => d.max().map(|x| DataType::duration_min(*x)),
+            _ => None,
+        }
+    }
+
+    pub fn upper_bound(&self) -> Option<DataType> {
```

Comment:
I think you are effectively implementing the `super_image` of the `minimum(x,y)` `Function` when x is a Column and y an expression.
It would be nice to benefit from all the automatic conversions by simply implementing `minimum(x,y)` and `maximum(x,y)`.